astropy/coordinates/representation.py

```diff
@@ -634,13 +634,13 @@ def _re_represent_differentials(self, new_rep, differential_class):
                 list(self.differentials.keys())[0]: differential_class
             }
 
-        elif set(differential_class.keys()) != set(self.differentials.keys()):
-            ValueError("Desired differential classes must be passed in "
-                       "as a dictionary with keys equal to a string "
-                       "representation of the unit of the derivative "
-                       "for each differential stored with this "
-                       "representation object ({0})"
-                       .format(self.differentials))
+        elif set(differential_class) != set(self.differentials):
```

Comment:
I think this could instead be:
```suggestion
        elif differential_class.keys() != self.differentials.keys():
```

Reply:
I have made the suggested change of comparison using keys. I think that I will not be able to add a doctest.